starlite/middleware/session.py

```diff
@@ -0,0 +1,233 @@
+    """Controls whether or not a cookie is sent with cross-site requests. Defaults to 'lax'."""
+
+    @validator("secret", always=True)
+    def validate_secret(cls, value: SecretBytes) -> SecretBytes:  # pylint: disable=no-self-argument
+        """Ensures that the 'secret' value is 128, 192 or 256 bits.
+
+        Args:
+            value: A bytes string.
+
+        Raises:
+            ValueError: if the bytes string is of incorrect length.
+
+        Returns:
+            A bytes string.
+        """
+        if len(value.get_secret_value()) not in [16, 24, 32]:
+            raise ValueError("secret length must be 16 (128 bit), 24 (192 bit) or 32 (256 bit)")
+        return value
+
+
+class SessionMiddleware(MiddlewareProtocol):
+    def __init__(
+        self,
+        app: "ASGIApp",
+        config: SessionCookieConfig,
+    ):
+        """Starlite SessionMiddleware.
+
+        Args:
+            app: The 'next' ASGI app to call.
+            config: SessionCookieConfig instance.
+        """
+        super().__init__(app)
+        self.app = app
+        self.config = config
+        self.aesgcm = AESGCM(config.secret.get_secret_value())
+
+    def dump_data(self, data: Any) -> List[bytes]:
+        """Given orjson serializable data, including pydantic models and numpy
+        types, dump it into a bytes string, encrypt, encode and split it into
+        chunks of the desirable size.
+
+        Args:
+            data: Data to serialize, encrypt, encode and chunk.
+
+        Notes:
+            - The returned list is composed of a chunks of a single base64 encoded
+                string that is encrypted using AES-CGM.
+
+        Returns:
+            List of encoded bytes string of a maximum length equal to the 'CHUNK_SIZE' constant.
+        """
+        serialized = dumps(data, default=Response.serializer, option=OPT_SERIALIZE_NUMPY)
+        aad = dumps({"time_to_live": round(time.time()) + self.config.max_age})
+        nonce = urandom(NONCE_SIZE)
+        encrypted = self.aesgcm.encrypt(nonce, serialized, associated_data=aad)
+        encoded = b64encode(nonce + encrypted + AAD + aad)
+        return [encoded[i : i + CHUNK_SIZE] for i in range(0, len(encoded), CHUNK_SIZE)]
+
+    def load_data(self, data: List[bytes]) -> Any:
+        """Given a list of strings, decodes them into the session object.
+
+        Args:
+            data: A list of strings derived from the request's session cookie(s).
+
+        Returns:
+            A deserialized session value.
+        """
+        decoded = b64decode(b"".join(data))
+        nonce = decoded[:NONCE_SIZE]
+
+        aad = None
+        aad_footer = decoded.find(AAD)
+        if aad_footer != -1:
+            aad = decoded[aad_footer:].replace(AAD, b"")
+
+        encrypted_session = decoded[NONCE_SIZE:aad_footer]
+        decrypted = self.aesgcm.decrypt(nonce, encrypted_session, associated_data=aad)
+
+        session_validation = loads(aad)["time_to_live"] > round(time.time())
```

Comment:
better to split this into a variable assignment:

```python
aad_data = loads(aad)
return loads(decrypted if aad_data and aad_data["expiration"] >= round(time.time())
```

Reply:
No changes for this. It's not possible for associated data to be None. The previous line will raise the Invalid Tag exception if AAD has been removed/tampered by the attacker.

```python
session_validation = loads(associated_data)["expires_at"] > round(time.time())
return loads(decrypted) if session_validation else {}
```